### PR TITLE
add datetime picker to dvl-core

### LINKED ISSUE #112: Include date and time picker in style guide

### PR DESCRIPTION
**Pull request by ajb**
closes #112 

img

### CODE REVIEW

File: app/inputs/datetime_picker_input.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  def initialize(*args)
+    super(*args)
+
+    # Format value as iso8601 for javascript new Date() compatibility
+    if object && (value = object.send(attribute_name))
+      input_html_options[:value] = value.iso8601
+    end
+  end
+
+  def input(wrapper_options)
```
Comment (dobthound) [inline app/inputs/datetime_picker_input.rb]: Unused method argument - `wrapper_options`. If it's necessary, use `_` or `_wrapper_options` as an argument name to indicate that it won't be used. You can also write as `input(*)` if you want the method to accept any arguments but don't care about them.

File: app/inputs/datetime_picker_input.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    content_tag(:div, class: 'datetime_input_wrapper cf') do
+      @builder.hidden_field(attribute_name, input_html_options) +
+
+      content_tag(:div, class: 'input_group datetime_date_group') do
+        tag(:input, type: 'text') +
+        content_tag(:a, class: 'button small') do
+          content_tag :i, class: 'fa fa-calendar' do; end;
```
Comment (dobthound) [inline app/inputs/datetime_picker_input.rb]: Prefer {...} over do...end for single-line blocks.<br>Do not use semicolons to terminate expressions.

File: app/inputs/datetime_picker_input.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        end
+      end +
+
+      content_tag(:div, class: 'input_group datetime_time_group') do
+        tag(:input, type: 'text') +
+        content_tag(:a, class: 'button small') do
+          content_tag :i, class: 'fa fa-clock-o' do; end;
```
Comment (dobthound) [inline app/inputs/datetime_picker_input.rb]: Prefer {...} over do...end for single-line blocks.<br>Do not use semicolons to terminate expressions.

File: app/inputs/datetime_picker_input.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        end
+      end +
+
+      ( options[:clear_text] ?
```
Comment (dobthound) [inline app/inputs/datetime_picker_input.rb]: Avoid multi-line ?: (the ternary operator); use `if`/`unless` instead.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 5]
```diff
@@ -0,0 +1,1681 @@
...
+(function($, undefined){
+
+  var $window = $(window);
+
+  function UTCDate(){
+    return new Date(Date.UTC.apply(Date, arguments));
+  }
+  function UTCToday(){
+    var today = new Date();
+    return UTCDate(today.getFullYear(), today.getMonth(), today.getDate());
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Missing 'new' prefix when invoking a constructor.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 6]
```diff
@@ -0,0 +1,1681 @@
...
+    var extras = {
+      get: function(i){
+        return this.slice(i)[0];
+      },
+      contains: function(d){
+        // Array.indexOf is not cross-browser;
+        // $.inArray doesn't work with Dates
+        var val = d && d.valueOf();
+        for (var i=0, l=this.length; i < l; i++)
+          if (this[i].valueOf() === val)
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'if'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 7]
```diff
@@ -0,0 +1,1681 @@
...
+            return i;
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'return'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 8]
```diff
@@ -0,0 +1,1681 @@
...
+        return -1;
+      },
+      remove: function(i){
+        this.splice(i,1);
+      },
+      replace: function(new_array){
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier 'new_array' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 9]
```diff
@@ -0,0 +1,1681 @@
...
+        if (!new_array)
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier 'new_array' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 10]
```diff
@@ -0,0 +1,1681 @@
...
+          return;
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'return'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 11]
```diff
@@ -0,0 +1,1681 @@
...
+        if (!$.isArray(new_array))
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier 'new_array' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 12]
```diff
@@ -0,0 +1,1681 @@
...
+          new_array = [new_array];
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier 'new_array' is not in camel case.<br>Expected '{' and instead saw 'new_array'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 13]
```diff
@@ -0,0 +1,1681 @@
...
+        this.clear();
+        this.push.apply(this, new_array);
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier 'new_array' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 14]
```diff
@@ -0,0 +1,1681 @@
...
+      return a;
+    };
+  })();
+
+
+  // Picker object
+
+  var Datepicker = function(element, options){
+    this.dates = new DateArray();
+    this.viewDate = UTCToday();
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Missing 'new' prefix when invoking a constructor.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 15]
```diff
@@ -0,0 +1,1681 @@
...
+    this.focusDate = null;
+
+    this._process_options(options);
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier '_process_options' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 16]
```diff
@@ -0,0 +1,1681 @@
...
+
+    this.element = $(element);
+    this.isInline = false;
+    this.isInput = this.element.is('input');
+    this.component = this.element.is('.date') ? this.element.find('.add-on, .input-group-addon, .btn') : false;
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Line is too long.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 17]
```diff
@@ -0,0 +1,1681 @@
...
+    this.hasInput = this.component && this.element.find('input').length;
+    if (this.component && this.component.length === 0)
+      this.component = false;
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'this'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 18]
```diff
@@ -0,0 +1,1681 @@
...
+    }
+
+    if (this.o.rtl){
+      this.picker.addClass('datepicker-rtl');
+    }
+
+    this.viewMode = this.o.startView;
+
+    if (this.o.calendarWeeks)
+      this.picker.find('tfoot th.today, tfoot th.clear')
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'this'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 19]
```diff
@@ -0,0 +1,1681 @@
...
+            .attr('colspan', function(i, val){
+              return parseInt(val) + 1;
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Missing radix parameter.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 20]
```diff
@@ -0,0 +1,1681 @@
...
+            });
+
+    this._allow_update = false;
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier '_allow_update' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 21]
```diff
@@ -0,0 +1,1681 @@
...
+
+    this.setStartDate(this._o.startDate);
+    this.setEndDate(this._o.endDate);
+    this.setDaysOfWeekDisabled(this.o.daysOfWeekDisabled);
+
+    this.fillDow();
+    this.fillMonths();
+
+    this._allow_update = true;
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier '_allow_update' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 22]
```diff
@@ -0,0 +1,1681 @@
...
+
+    if (this.isInline){
+      this.show();
+    }
+  };
+
+  Datepicker.prototype = {
+    constructor: Datepicker,
+
+    _process_options: function(opts){
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier '_process_options' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 23]
```diff
@@ -0,0 +1,1681 @@
...
+      // Processed options
+      var o = this.o = $.extend({}, this._o);
+
+      // Check if "de-DE" style date is available, if not language should
+      // fallback to 2 letter code eg "de"
+      var lang = o.language;
+      if (!dates[lang]){
+        lang = lang.split('-')[0];
+        if (!dates[lang])
+          lang = defaults.language;
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'lang'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 24]
```diff
@@ -0,0 +1,1681 @@
...
+          o.minViewMode = 0;
+      }
+
+      o.startView = Math.max(o.startView, o.minViewMode);
+
+      // true, false, or Number > 0
+      if (o.multidate !== true){
+        o.multidate = Number(o.multidate) || false;
+        if (o.multidate !== false)
+          o.multidate = Math.max(0, o.multidate);
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'o'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 25]
```diff
@@ -0,0 +1,1681 @@
...
+        else
+          o.multidate = 1;
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'o'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 26]
```diff
@@ -0,0 +1,1681 @@
...
+      o.multidateSeparator = String(o.multidateSeparator);
+
+      o.weekStart %= 7;
+      o.weekEnd = ((o.weekStart + 6) % 7);
+
+      var format = DPGlobal.parseFormat(o.format);
+      if (o.startDate !== -Infinity){
+        if (!!o.startDate){
+          if (o.startDate instanceof Date)
+            o.startDate = this._local_to_utc(this._zero_time(o.startDate));
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'o'.<br>Identifier '_local_to_utc' is not in camel case.<br>Identifier '_zero_time' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 27]
```diff
@@ -0,0 +1,1681 @@
...
+          else
+            o.startDate = DPGlobal.parseDate(o.startDate, format, o.language);
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'o'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 28]
```diff
@@ -0,0 +1,1681 @@
...
+        }
+        else {
+          o.startDate = -Infinity;
+        }
+      }
+      if (o.endDate !== Infinity){
+        if (!!o.endDate){
+          if (o.endDate instanceof Date)
+            o.endDate = this._local_to_utc(this._zero_time(o.endDate));
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'o'.<br>Identifier '_local_to_utc' is not in camel case.<br>Identifier '_zero_time' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 29]
```diff
@@ -0,0 +1,1681 @@
...
+          else
+            o.endDate = DPGlobal.parseDate(o.endDate, format, o.language);
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'o'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 30]
```diff
@@ -0,0 +1,1681 @@
...
+        }
+        else {
+          o.endDate = Infinity;
+        }
+      }
+
+      o.daysOfWeekDisabled = o.daysOfWeekDisabled||[];
+      if (!$.isArray(o.daysOfWeekDisabled))
+        o.daysOfWeekDisabled = o.daysOfWeekDisabled.split(/[,\s]*/);
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'o'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 31]
```diff
@@ -0,0 +1,1681 @@
...
+      });
+
+      var plc = String(o.orientation).toLowerCase().split(/\s+/g),
+        _plc = o.orientation.toLowerCase();
+      plc = $.grep(plc, function(word){
+        return (/^auto|left|right|top|bottom$/).test(word);
+      });
+      o.orientation = {x: 'auto', y: 'auto'};
+      if (!_plc || _plc === 'auto')
+        ; // no action
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw ';'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 32]
```diff
@@ -0,0 +1,1681 @@
...
+      }
+    },
+    _buildEvents: function(){
+      if (this.isInput){ // single input
+        this._events = [
+          [this.element, {
+            focus: $.proxy(this.show, this),
+            keyup: $.proxy(function(e){
+              if ($.inArray(e.keyCode, [27,37,39,38,40,32,13,9]) === -1)
+                this.update();
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'this'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 33]
```diff
@@ -0,0 +1,1681 @@
...
+        ];
+      }
+      else if (this.component && this.hasInput){ // component: input + button
+        this._events = [
+          // For components that are not readonly, allow keyboard nav
+          [this.element.find('input'), {
+            focus: $.proxy(this.show, this),
+            keyup: $.proxy(function(e){
+              if ($.inArray(e.keyCode, [27,37,39,38,40,32,13,9]) === -1)
+                this.update();
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'this'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 34]
```diff
@@ -0,0 +1,1681 @@
...
+          [this.element, {
+            click: $.proxy(this.show, this)
+          }]
+        ];
+      }
+      this._events.push(
+        // Component: listen for blur on element descendants
+        [this.element, '*', {
+          blur: $.proxy(function(e){
+            this._focused_from = e.target;
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier '_focused_from' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 35]
```diff
@@ -0,0 +1,1681 @@
...
+          }, this)
+        }],
+        // Input: listen for blur on element
+        [this.element, {
+          blur: $.proxy(function(e){
+            this._focused_from = e.target;
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier '_focused_from' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 36]
```diff
@@ -0,0 +1,1681 @@
...
+    _attachSecondaryEvents: function(){
+      this._detachSecondaryEvents();
+      this._applyEvents(this._secondaryEvents);
+    },
+    _detachSecondaryEvents: function(){
+      this._unapplyEvents(this._secondaryEvents);
+    },
+    _trigger: function(event, altdate){
+      var date = altdate || this.dates.get(-1),
+        local_date = this._utc_to_local(date);
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier 'local_date' is not in camel case.<br>Identifier '_utc_to_local' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 37]
```diff
@@ -0,0 +1,1681 @@
...
+
+      this.element.trigger({
+        type: event,
+        date: local_date,
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier 'local_date' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 38]
```diff
@@ -0,0 +1,1681 @@
...
+        dates: $.map(this.dates, this._utc_to_local),
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier '_utc_to_local' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 39]
```diff
@@ -0,0 +1,1681 @@
...
+          format = format || this.o.format;
+          var date = this.dates.get(ix);
+          return DPGlobal.formatDate(date, format, this.o.language);
+        }, this)
+      });
+    },
+
+    show: function(){
+      if (!this.isInline)
+        this.picker.appendTo('body');
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'this'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 40]
```diff
@@ -0,0 +1,1681 @@
...
+      this.picker.show();
+      this.place();
+      this._attachSecondaryEvents();
+      this._trigger('show');
+    },
+
+    hide: function(){
+      if (this.isInline)
+        return;
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'return'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 41]
```diff
@@ -0,0 +1,1681 @@
...
+      if (!this.picker.is(':visible'))
+        return;
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'return'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 42]
```diff
@@ -0,0 +1,1681 @@
...
+      this.showMode();
+
+      if (
+        this.o.forceParse &&
+        (
+          this.isInput && this.element.val() ||
+          this.hasInput && this.element.find('input').val()
+        )
+      )
+        this.setValue();
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Expected '{' and instead saw 'this'.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 43]
```diff
@@ -0,0 +1,1681 @@
...
+      this._detachEvents();
+      this._detachSecondaryEvents();
+      this.picker.remove();
+      delete this.element.data().datepicker;
+      if (!this.isInput){
+        delete this.element.data().date;
+      }
+    },
+
+    _utc_to_local: function(utc){
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier '_utc_to_local' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 44]
```diff
@@ -0,0 +1,1681 @@
...
+      return utc && new Date(utc.getTime() + (utc.getTimezoneOffset()*60000));
+    },
+    _local_to_utc: function(local){
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier '_local_to_utc' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 45]
```diff
@@ -0,0 +1,1681 @@
...
+      return local && new Date(local.getTime() - (local.getTimezoneOffset()*60000));
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Line is too long.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 46]
```diff
@@ -0,0 +1,1681 @@
...
+    },
+    _zero_time: function(local){
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Identifier '_zero_time' is not in camel case.

File: vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js [hunk 47]
```diff
@@ -0,0 +1,1681 @@
...
+      return local && new Date(local.getFullYear(), local.getMonth(), local.getDate());
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/bootstrap-datepicker.js]: Line is too long.<br>Too many errors. (27% scanned).

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 48]
```diff
@@ -0,0 +1,1151 @@
...
+requires jQuery 1.7+
+************************/
+
+
+(function (factory) {
+    if (typeof exports === "object" && exports &&
+        typeof module === "object" && module && module.exports === exports) {
+        // Browserify. Attach to jQuery module.
+        factory(require("jquery"));
+    } else if (typeof define === 'function' && define.amd) {
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 49]
```diff
@@ -0,0 +1,1151 @@
...
+    // AMD. Register as an anonymous module.
+    define(['jquery'], factory);
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 50]
```diff
@@ -0,0 +1,1151 @@
...
+  } else {
+    // Browser globals
+    factory(jQuery);
+  }
+}(function ($) {
+  var _baseDate = _generateBaseDate();
+  var _ONE_DAY = 86400;
+  var _lang = {
+    am: 'am',
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 51]
```diff
@@ -0,0 +1,1151 @@
...
+    pm: 'pm',
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 52]
```diff
@@ -0,0 +1,1151 @@
...
+    AM: 'AM',
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 53]
```diff
@@ -0,0 +1,1151 @@
...
+    PM: 'PM',
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 54]
```diff
@@ -0,0 +1,1151 @@
...
+    decimal: '.',
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 55]
```diff
@@ -0,0 +1,1151 @@
...
+    mins: 'mins',
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 56]
```diff
@@ -0,0 +1,1151 @@
...
+    hr: 'hr',
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 57]
```diff
@@ -0,0 +1,1151 @@
...
+    hrs: 'hrs'
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 58]
```diff
@@ -0,0 +1,1151 @@
...
+
+        // pick up settings from data attributes
+        var attributeOptions = [];
+        for (var key in $.fn.timepicker.defaults) {
+          if (self.data(key))  {
+            attributeOptions[key] = self.data(key);
+          }
+        }
+
+        var settings = $.extend({}, $.fn.timepicker.defaults, attributeOptions, options);
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Line is too long.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 59]
```diff
@@ -0,0 +1,1151 @@
...
+
+        if (settings.lang) {
+          _lang = $.extend(_lang, settings.lang);
+        }
+
+        settings = _parseSettings(settings);
+        self.data('timepicker-settings', settings);
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 60]
```diff
@@ -0,0 +1,1151 @@
...
+        self.addClass('ui-timepicker-input');
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 61]
```diff
@@ -0,0 +1,1151 @@
...
+
+        if (settings.useSelect) {
+          _render(self);
+        } else {
+          self.prop('autocomplete', 'off');
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 62]
```diff
@@ -0,0 +1,1151 @@
...
+          self.on('click.timepicker focus.timepicker', methods.show);
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 63]
```diff
@@ -0,0 +1,1151 @@
...
+          self.on('change.timepicker', _formatValue);
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 64]
```diff
@@ -0,0 +1,1151 @@
...
+          self.on('keydown.timepicker', _keydownhandler);
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 65]
```diff
@@ -0,0 +1,1151 @@
...
+          self.on('keyup.timepicker', _keyuphandler);
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 66]
```diff
@@ -0,0 +1,1151 @@
...
+
+          _formatValue.call(self.get(0));
+        }
+      });
+    },
+
+    show: function(e)
+    {
+      var self = $(this);
+      var settings = self.data('timepicker-settings');
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 67]
```diff
@@ -0,0 +1,1151 @@
...
+      if (e) {
+        if (!settings.showOnFocus) {
+          return true;
+        }
+
+        e.preventDefault();
+      }
+
+      if (settings.useSelect) {
+        self.data('timepicker-list').focus();
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 68]
```diff
@@ -0,0 +1,1151 @@
...
+        return;
+      }
+
+      if (_hideKeyboard(self)) {
+        // block the keyboard on mobile devices
+        self.blur();
+      }
+
+      var list = self.data('timepicker-list');
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 69]
```diff
@@ -0,0 +1,1151 @@
...
+
+      // check if input is readonly
+      if (self.prop('readonly')) {
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 70]
```diff
@@ -0,0 +1,1151 @@
...
+        return;
+      }
+
+      // check if list needs to be rendered
+      if (!list || list.length === 0 || typeof settings.durationTime === 'function') {
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Line is too long.<br>Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 71]
```diff
@@ -0,0 +1,1151 @@
...
+        _render(self);
+        list = self.data('timepicker-list');
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

File: vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js [hunk 72]
```diff
@@ -0,0 +1,1151 @@
...
+      }
+
+      // make sure other pickers are hidden
+      methods.hide();
+
+      // position the dropdown relative to the input
+      list.show();
+      var listOffset = {};
+
+      if (settings.orientation == 'rtl') {
```
Comment (dobthound) [inline vendor/assets/javascripts/dvl/vendor/jquery.timepicker.js]: Mixed double and single quotes.

**Comment by jrubenoff** [comment]
Standardized the styles here a bit.

screen shot 2015-07-15 at 10 45 07 am

screen shot 2015-07-15 at 10 45 22 am

What else needs review before merging?

**Comment by ajb** [comment]
Looks great, I just need to review those bits.

On Wed, Jul 15, 2015 at 1:46 PM, Josh Rubenoff <notifications@github.com>
wrote:

> Standardized the styles here a bit.
>
> [image: screen shot 2015-07-15 at 10 45 07 am]
> <https://cloud.githubusercontent.com/assets/1328849/8705476/b1920436-2ade-11e5-9a3b-a5dc034b8178.png>
>
> [image: screen shot 2015-07-15 at 10 45 22 am]
> <https://cloud.githubusercontent.com/assets/1328849/8705473/adfe3c40-2ade-11e5-9c39-91659c69b60a.png>
>
> What else needs review before merging?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dobtco/dvl-core/pull/121#issuecomment-121690904>.
>



-- 
Adam Becker
(951) 9-BECKER
@AdamJacobBecker

**Comment by ajb** [comment]
Looks good. The hacks for the `.next` and `.prev` buttons were causing issues, so I just made the change in the Javascript instead.

**Pull request merged**